3 - PB/Documentazione esterna/Specifica Tecnica/Specifica Tecnica.typ

```diff
@@ -146,7 +147,199 @@ Ciò avviene prestando attenzione alle tecnologie impiegate sia nel front-end ch
+
+- *pattern Repository*: come per il precedente pattern, viene implementato per separare la logica di business dalla logica di accesso ai dati. Le classi relative a questo pattern eseguono operazioni di lettura, aumentando l'astrazione dei dettagli specifici della persistenza dei dati e permettendo all'applicazione di interagirvi in modo indipendente dal tipo di archivio sottostante;
+
+- *pattern Provider*: applicato nel contesto tecnologico del progetto, soprattutto rispetto all'utilizzo di React, vengono sfruttate delle API per permette di gestire e trasferire i dati attraverso l'albero dei componenti in modo strutturato, evitando il "prop drilling", ovvero l'effetto che si verifica nei casi in cui è necessario trasportare i dati attraverso più livelli di componenti, anche se alcuni di essi non ne necessitano;
+
+- *pattern Strategy*: consente di definire una famiglia di algoritmi, incapsularli in classi separate e renderli intercambiabili. In questo modo è possibile applicare l'algoritmo appropriato senza dover conoscere i dettagli implementativi.
+
+
+
+== Classi e Componenti
+Ciascun layer possiede il suo indipendente sistema di classi e componenti e prevede metodi per comunicare con i layer adiacenti.
+
+=== Persistence layer
+Mediante delle chiamate API vengono letti i dati utili all'applicazione da un database esterno e vengono forniti degli adeguati file Json.
+Questi ultimi vengono utilizzati dalle classi derivate dall'adozione del pattern "Data mapper":
+- l'interfaccia DataMapperInterface;
+- binMapper: implementa DataMapperInterface ed è responsabile della creazione di oggetti "bin";
+- productMapper: implementa DataMapperInterface ed è responsabile della creazione di oggetti "product";
+- zoneMapper: implementa DataMapperInterface ed è responsabile della creazione di oggetti "zone".
+
+
+Al fine di dividere ulteriormente il Persistence layer dal Business layer, viene applicato il pattern "Repository" mediante le classi:
+- l'interfaccia DataRepositoryInterface;
+- binRepository: implementa DataRepositoryInterface ed è responsabile delle chiamate API relative agli oggetti "bin";
+- productRepository: implementa DataRepositoryInterface ed è responsabile delle chiamate API relative agli oggetti "product";
+- zoneRepository: implementa DataRepositoryInterface ed è responsabile delle chiamate API relative agli oggetti "zone".
+
+Queste classi utilizzano le corrispondenti classi del pattern Data mapper.
+
+
+
+#figure(
+  image("./imgs/Persistence Layer patterns class diagram.png", width: 100%),
+  caption: [
+    Diagramma delle classi del layer di persistenza
+  ],
+)
+
+=== Business layer
```

Comment:
Negli elenchi di questa sezione non so se sia meglio utilizzare monospace o grassetto, perché sono sia nomi di attributi che termini da definire ([Norme](https://error-418-swe.github.io/Documenti/3%20-%20PB/Documentazione%20interna/Norme%20di%20Progetto_v1.24.0.pdf) Sezione 4.6.4.2.2) @Carraro-Riccardo te cosa dici?

Reply:
attualmente ci sono liste con elementi in monospace e in grassetto. Mi direte quale standard è più corretto